### PR TITLE
Added option to disable UPnP port forwarding

### PR DESCRIPTION
**Pull request by Measurity**

### CODE REVIEW

File: NitroxServer/Communication/PortForward.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                    catch (TaskCanceledException)
+                    {
+                        // ignored
+                    }
+                }
+            }
+            return false;
+        }
+
+        private static event EventHandler<INatDevice> DeviceDiscovered;
```
Comment (Coding-Hen) [inline NitroxServer/Communication/PortForward.cs:90]: Can we move this to the top of the class with the other properties

**Review by Coding-Hen — CHANGES_REQUESTED** [review]

File: NitroxServer/Communication/PortForward.cs (hunk 1 above)
Comment (Measurity) [reply NitroxServer/Communication/PortForward.cs:90]: Back to fighting the code style again :(

Comment (killzoms) [reply NitroxServer/Communication/PortForward.cs:90]: lol

File: NitroxServer/Communication/PortForward.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        /// <summary>
+        ///     Tries to open the given port.
+        /// </summary>
+        public static async Task<bool> TryOpenPortAsync(int port, TimeSpan timeout)
+        {
+            static async Task PortForwardDeviceAsync(IDictionary<int, Exception> errors, INatDevice device, int port)
+            {
+                try
+                {
+                    await device.CreatePortMapAsync(new Mapping(Protocol.Udp, port, port, (int)TimeSpan.FromDays(1).TotalSeconds, "Nitrox Server - Subnautica"));
```
Comment (Jannify) [inline NitroxServer/Communication/PortForward.cs:25]: As it's in the NitroxServer namespace it should probably only be "Nitrox Server".

**Review by Jannify — APPROVED** [review]

**Pull request merged**